celltype_annotation/analysis/cell-assign-sarcoma.Rmd

```diff
@@ -0,0 +1,451 @@
+The filtered list of cell type markers is turned into a binary gene by cell type matrix with an additional "other" column where no marker genes are found. 
+This is used as input to run `CellAssign` using `scripts/cell-assign.py`. 
+The output of that script is a prediction matrix where each row is a cell barcode and each column is a cell type found in the reference matrix. 
+The corresponding cell type with the top score for each barcode is then assigned to that cell. 
+
+## Set Up
+
+```{r}
+library(SingleCellExperiment)
+library(ggplot2)
+```
+
+```{r}
+# Set up paths 
+
+# build path to annotated sce file 
+processed_data_dir <-file.path("..", "data") 
+processed_sce_file <- glue::glue("{params$library_id}_processed_celltype.rds")
+local_processed_sce_path <- file.path(processed_data_dir, params$sample_id, processed_sce_file)
+
+# anndata output
+anndata_dir <- file.path("..", "data", "anndata")
+anndata_file <- glue::glue("{params$library_id}_anndata.hdf5")
+anndata_file <- file.path(anndata_dir, anndata_file)
+fs::dir_create(anndata_dir)
+
+# reference files for cell marker
+ref_dir <- file.path("..", "references")
+excel_celltype_file <- file.path(ref_dir, "Cell_marker_Human.xlsx")
+panglao_file <- file.path(ref_dir, "PanglaoDB_markers_27_Mar_2020.tsv")
+
+# output matrix files 
+cell_marker_mtx_file <- file.path(ref_dir, "cell_marker_muscle_mtx.csv")
+panglao_matrix_file <- file.path(ref_dir, "panglao_connective_tissue_mtx.csv")
+
+# cellassign predictions
+cellmarker_prediction_file <- file.path(
+  anndata_dir, 
+  glue::glue("{params$library_id}_muscle.tsv")
+)
+panglao_prediction_file <- file.path(
+  anndata_dir, 
+  glue::glue("{params$library_id}_panglao_connective_tissue.tsv")
+)
+
+```
+
+
+```{r}
+# read in all annotated sce 
+processed_sce <- readr::read_rds(local_processed_sce_path)
+```
+
+```{r}
+# write out anndata 
+scpcaTools::sce_to_anndata(processed_sce, anndata_file = anndata_file)
+```
+
+```{r}
+# source helper functions
+function_file <- file.path("..", "utils", "cellassign-helper-functions.R")
+source(function_file)
+```
+
+## CellMarker2.0
+
+```{r}
+# read in excel file
+celltype_markers_df <- openxlsx::read.xlsx(excel_celltype_file)
+```
+
+
+```{r}
+# look at different tissue types available 
+celltype_markers_df |>
+  dplyr::count(tissue_class)
+```
+
+```{r}
+celltype_markers_df |> 
+  dplyr::count(cancer_type, cellontology_id, cell_name) |>
+  dplyr::mutate(sarcoma_filter = stringr::str_detect(cancer_type, "sarcoma")) |>
+  dplyr::filter(sarcoma_filter)
+```
+
+
+```{r}
+# grab the muscle tissue type
+muscle_markers_df <- celltype_markers_df |> 
+  dplyr::filter(tissue_class == "Muscle") |>
+  tidyr::drop_na(cellontology_id, Symbol)
+
+# print total number of marker genes for each cellontology id
+muscle_markers_df |>
+  dplyr::count(cell_name, cancer_type, cellontology_id)
```

Comment:
Okay, so what does it mean to divide a cell type between cancer types here? Is the single marker of myoblasts in rhabdomyosarcoma a subset of the 35 in normal tissue? Or is it in addition to the normal markers? Or is it from a different set of experiments? (Same question for stem cells and presumably every other cell type, but that was the first one that really jumped out to me)

The answers to these questions may not all be important, but might affect if/how we would choose to use this resource.

(Also confused by skeletal muscle not showing up in normal)